core/src/com/unciv/models/ruleset/unique/UniqueType.kt

```diff
@@ -755,6 +755,9 @@ enum class UniqueType(
     SkipPromotion("Doing so will consume this opportunity to choose a Promotion", UniqueTarget.Promotion),
     FreePromotion("This Promotion is free", UniqueTarget.Promotion),
 
+    OneTimeCreateTerrainFeatureImprovement("Create a [terrainFeature/improvementName] on this tile", UniqueTarget.UnitTriggerable),
```

Comment:
Makes `ConstructImprovementInstantly` redundant? Tbh, I think some of the unit actions should probably become triggers anyways. Probably should deprecate?

Reply:
To do that, we'll need to add a way to limit this construction according to the improvement's 'can be constructed on' rules

Reply:
To be fair, I'm not sure this unique implies it should ignore limitations